xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -402,6 +410,81 @@ StatusOr<bool> IsSupportedBMM2(const HloInstruction* bmm_2,
   return true;
 }
 
+StatusOr<bool> IsFlashAttention(HloInstruction* bmm_1, bool is_causal_mask,
+                                absl::string_view custom_call_name) {
+  TF_ASSIGN_OR_RETURN(
+      std::vector<int64_t> seq_q_dims,
+      GetNonContractingDims(
+          bmm_1->operand(0)->shape(),
+          bmm_1->dot_dimension_numbers().lhs_batch_dimensions(),
```

Comment:
I think it would be useful to use const reference variable to bmm_1->dot_dimension_numbers(), as this is used quite often in the following code.